### ISSUE TITLE
Issue running bcftools view --trim-alt-alleles

### ISSUE THREAD

**Issue by niruchennagiri-claritas** [samtools/bcftools#322]
I am running trim-alt-alleles using the command:
bcftools view --trim-alt-alleles input.vcf

I get the error:
chr1    59348Assertion failed: (nret==nA_ori), function bcf_remove_alleles, file vcfutils.c, line 333.

The line from input vcf file this errors on is:
chr1    5934837 .   C   T   1117.63 PASS    AF=0.976378;AO=121;DP=131;FAO=124;FDP=127;FR=.;FRO=3;FSAF=50;FSAR=74;FSRF=2;FSRR=1;FWDB=0.0241805;FXX=0.030532;HRUN=4;LEN=1;MLLD=128.841;OALT=T;OID=.;OMAPALT=T;OPOS=5934837;OREF=C;PB=0.5;PBP=1;QD=35.201;RBI=0.0514354;REFB=0.00491583;REVB=0.0453972;RO=2;SAF=50;SAR=71;SRF=1;SRR=1;SSEN=0;SSEP=0;SSSB=-0.00276194;STB=0.506448;STBP=0.411;TYPE=snp;VARB=0.00123751  GT:AF:AO:DP:FAO:FDP:FRO:FSAF:FSAR:FSRF:FSRR:GQ:RO:SAF:SAR:SRF:SRR   1/1:0.976378:121:131:124:127:3:50:74:2:1:33:2:50:71:1:1

**Comment by jrandall** [comment]
@niruchennagiri-claritas could you possibly post a minimal example of this bug (i.e. a VCF file including the header and the problematic line) so that we can test it against the current development version?

**Comment by macsx82** [comment]
Hi Josh,

I get the same error

bcftools: vcfutils.c:358: bcf_remove_allele_set: Assertion `nret==nA_ori' failed

with the following:

bcftools view -c1 ${file} | bcftools view -a -O z -o ${basedir}/TRIMMED/M1.${filename}

(which I tried also in another form :
bcftools view -c1 view -a -O z -o ${basedir}/TRIMMED/M1.${filename} ${file} )

If I run the command on a subset which includes the last line of the truncated file generated and a few lines before and after I don't get any error, though (file attached).
I just realized I don't get any error adding the -G option neither .
bcftools view -G -c1 ${file} | bcftools view -a -O z -o ${basedir}/TRIMMED/M1.${filename}

Strange thing is, this is happening only for one out of three cohorts I'm working on and on all chromosomes but chr 18, 21, 22.

Thanks for your help,
Max

[TEST.8.vcf.txt](https://github.com/samtools/bcftools/files/108273/TEST.8.vcf.txt)

**Comment by pd3** [comment]
Hi Max, I am not able to reproduce the error with the file you provided. What version of bcftools are you using? Can you please try with the latest version?

**Comment by macsx82** [comment]
Hi Petr,

I'm working with v.1.3 but I get the same error with v.1.2 . Since I'm having this problem with only a cohort out of three (I processed them in the same way) I was wondering whether my files were in some way corrupted.
I checked and the error is triggered only after a step of annotation using GATK, for some chromosomes (for example, I'm able to trim chr 1 and 2 but not 3 and 4). If I check the files generated after a subsequent step of annotation using VEP, I can only trim chr 18, 21 and 22 (as reported in the previous message). I had a look at the row pointed out in the error message (358, vcfutils.c) but I can't understand if the problem it's something related to the genotypes or the file header (but I need to have a better look at the code, to better understand the function).
I was able to replicate the error with this command, on the attached file (remove the .txt extension) :
bcftools view -a -r 2:13949-5013948 M3.tt.2.vcf.gz -O z -o test.vcf.gz

thanks a lot,
Max

[M3.tt.2.vcf.gz.txt](https://github.com/samtools/bcftools/files/115791/M3.tt.2.vcf.gz.txt)
